invenio_rest/views.py

```diff
@@ -80,10 +87,25 @@ def make_response(self, *args, **kwargs):
         :raises :py:class:`werkzeug.exceptions.NotAcceptable`: if no media type
         matches current Accept header.
         """
-        best = request.accept_mimetypes.best_match(self.serializers.keys())
-        if best is None:
-            abort(406)
-        return self.serializers[best](*args, **kwargs)
+        # First search for a local serializer if one has been set
+        if self.method_serializers:
```

Comment:
IMHO my proposal is doing exactly same thing. Do you have any case on which it behaves differently?